lib/housekeeping.js

```diff
@@ -37,18 +38,148 @@ class HouseKeeper {
     this.monitor = monitor;
     this.tagger = tagger;
   }
+  
+  _calculateVolumeTotals(tabulatedVolumes) {
+    let totalVolumes = {};
+    
+    Object.keys(tabulatedVolumes).forEach(function(volumesInRegion) {
+      tabulatedVolumes[volumesInRegion].forEach(function(volume) {        
+        let region = volume.AvailabilityZone;
+        let type = volume.VolumeType;
+        
+        if (!(region in totalVolumes)) {
+          totalVolumes[region] = {};
```

Comment:
I'm wondering if we shouldn't just pass in a single list containing all volumes over all regions, since you're checking the region of all volume anyway. The alternative would be to first initialize the `totalVolumes` object with each of the keys in `tabulatedVolumes`, so you don't have to check if `totalVolumes` contains the region yet.

Side note: you don't need to check `volume.AvailabilityZone`, since `volumesInRegion` is actually the region name. Might be worth renaming that variable to reflect that it is the region name?

Reply:
Good catch. It's probably best to initialize the `totalVolumes` object with all of the regions.

Ahhhh... Right. I was wondering why I had to use `volume.AvailabilityZone`. I was blanking that I had `volumesInRegion`. You're right. My variable names need to be more descriptive/accurate.

Reply:
We've discussed so many times now how much both of us struggle with variable names! I really get a kick out of the fact that we can both do the hard part of coding and get stuck on naming! :rofl:

Reply:
I think @jhford was saying on IRC that naming's one of the hardest problems in CS. At least we're not the only ones!